campaign.go

```diff
@@ -0,0 +1,77 @@
+package opslevel
+
+type ListCampaignsVariables struct {
+	After  *string
+	First  *int
+	SortBy *CampaignSortEnum
+	Status *CampaignStatusEnum
+}
+
+func (v *ListCampaignsVariables) AsPayloadVariables() *PayloadVariables {
+	variables := PayloadVariables{}
+	if v.After != nil {
+		variables["after"] = *v.After
+	}
+	if v.First != nil {
+		variables["first"] = *v.First
+	}
+	if v.SortBy != nil {
+		variables["sortBy"] = *v.SortBy
+	}
+	if v.Status != nil {
+		// cast status to match filter argument type
+		variables["status"] = string(*v.Status)
+	}
+	return &variables
+}
+
+func (client *Client) ListCampaigns(campaignVariables *ListCampaignsVariables) (*CampaignConnection, error) {
+	if campaignVariables == nil {
+		campaignVariables = &ListCampaignsVariables{}
+	}
+
+	defaultPages := client.InitialPageVariablesPointer()
+
+	if campaignVariables.First == nil {
+		defaultFirst := (*defaultPages)["first"].(int)
+		campaignVariables.First = &defaultFirst
+	}
+
+	if campaignVariables.After == nil {
+		defaultAfter := (*defaultPages)["after"].(string)
+		campaignVariables.After = &defaultAfter
+	}
+
+	if campaignVariables.SortBy == nil {
+		campaignVariables.SortBy = &CampaignSortEnumStartDateDesc
+	}
+
+	if campaignVariables.Status == nil {
+		campaignVariables.Status = &CampaignStatusEnumInProgress
```

Comment:
If the customer didn't supply "status" and since there is no default value for it - my understanding is that the default case of using this function will fail - IE `client. ListCampaigns(nil)` will result in

```
query CampaignsList {
  account {
    campaigns(
      first: 100
      after: ""
      sortBy: null
      filter: [{key: status, arg: ""}]
    ) {
    ```
    
    and this fails for me with
    
    ```
    {
  "errors": [
    {
      "message": "invalid filter_params",
      "locations": [
        {
          "line": 3,
          "column": 5
        }
      ],
      "path": [
        "account",
        "campaigns"
      ]
    }
  ],
  "data": null
}
    ```